src/contrib/dependencyinjection/Akka.DependencyInjection/ServiceProviderDependencyResolver.cs

```diff
@@ -0,0 +1,60 @@
+// //-----------------------------------------------------------------------
+// // <copyright file="ServiceProviderDependencyResolver.cs" company="Akka.NET Project">
+// //     Copyright (C) 2009-2021 Lightbend Inc. <http://www.lightbend.com>
+// //     Copyright (C) 2013-2021 .NET Foundation <https://github.com/akkadotnet/akka.net>
+// // </copyright>
+// //-----------------------------------------------------------------------
+
+using System;
+using Akka.Actor;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Akka.DependencyInjection
+{
+    /// <summary>
+    /// INTERNAL API.
+    ///
+    /// <see cref="IDependencyResolver"/> implementation backed by <see cref="IServiceProvider"/>
+    /// </summary>
+    public class ServiceProviderDependencyResolver : IDependencyResolver
+    {
+        public IServiceProvider ServiceProvider { get; }
+
+        public ServiceProviderDependencyResolver(IServiceProvider serviceProvider)
+        {
+            ServiceProvider = serviceProvider;
+        }
+
+        public IResolverScope CreateScope()
+        {
+            return new ServiceProviderScope(ServiceProvider.CreateScope());
+        }
+        
+        public object GetService<T>()
+        {
+            return ServiceProvider.GetService<T>();
+        }
+        
+        public object GetService(Type type)
+        {
+            return ServiceProvider.GetService(type);
+        }
+        
+        public Props Props(Type type, params object[] args)
+        {
+            return Akka.Actor.Props.CreateBy(new ServiceProviderActorProducer(ServiceProvider, type, args));
+        }
+        
+        public Props Props(Type type)
+        {
+            return Akka.Actor.Props.CreateBy(new ServiceProviderActorProducer(ServiceProvider, type, Array.Empty<object>()));
```

Comment:
Its probably better if this calls Props(Type, params object[]) and have a type check using `Type.IsAssignableTo` to make sure that the type passed in actually inherits `ActorBase` (type safety)

Reply:
Good idea

Reply:
Resolved